homeassistant/components/mqtt/__init__.py

```diff
@@ -53,25 +54,62 @@
 ATTR_TOPIC = 'topic'
 ATTR_PAYLOAD = 'payload'
 ATTR_PAYLOAD_TEMPLATE = 'payload_template'
-ATTR_QOS = 'qos'
+ATTR_QOS = CONF_QOS
 ATTR_RETAIN = 'retain'
 
 MAX_RECONNECT_WAIT = 300  # seconds
 
+_VALID_QOS_SCHEMA = vol.All(vol.Coerce(int), vol.In([0, 1, 2]))
```

Comment:
I guess `vol.In()` and `vol.Any()` are similar. I like the `vol.In` in this case as it should be faster.